networkx/algorithms/shortest_paths/generic.py

```diff
@@ -294,23 +303,20 @@ def average_shortest_path_length(G, weight=None):
     1.0
 
     """
-    if G.is_directed():
-        if not nx.is_weakly_connected(G):
-            raise nx.NetworkXError("Graph is not connected.")
-    else:
-        if not nx.is_connected(G):
-            raise nx.NetworkXError("Graph is not connected.")
-    avg=0.0
-    if weight is None:
-        for node in G:
-            path_length=nx.single_source_shortest_path_length(G, node)
-            avg += sum(dist for n, dist in path_length)
-    else:
-        for node in G:
-            path_length=nx.single_source_dijkstra_path_length(G, node, weight=weight)
-            avg += sum(dist for n, dist in path_length)
-    n=len(G)
-    return avg/(n*(n-1))
+    # For the special case of the trivial graph, return zero immediately.
```

Comment:
My mistake--trivial graph has one node...  but still. We should include the corner case ```n=0``` too.